### PR TITLE
fix: fastlog PlayerAttackEvent::mAttackDamage; mod onHopperSearchItem onHopperPushOut onAttackEntity

### LINKED ISSUE #1027: `fastLog`行为异常
### 异常模块

ScriptEngine (脚本引擎)

### 操作系统

Windows Server 2022

### LiteLoader 版本

2.9.2-

### BDS 版本

1.19.51.01

### 发生了什么?

`fastLog`会多输出一个换行符

### 复现此问题的步骤

使用`fastLog`

### 有关的日志/输出

_No response_

### 插件列表

_No response_

### PR DESCRIPTION
**Pull request by cryx233**
## Description

fix: 
fastlog error, wrong PlayerAttackEvent::mAttackDamage

feat: 
add item object to ` onHopperSearchItem `, ` onHopperPushOut ` event
add damage argument to llse ` onAttackEntity ` event

## Issues fixed

Close #1027 #1050 #990

### CODE REVIEW

File: LiteLoader/include/llapi/EventAPI.h [hunk 1]
```diff
@@ -285,7 +282,7 @@ class PlayerAttackEvent : public EventTemplate<PlayerAttackEvent> {
 public:
     Player* mPlayer = nullptr;
     Actor* mTarget = nullptr;
-    int mAttackDamage = false;
+    float mAttackDamage = 0;
```
Comment (KawaiiNahida) [inline LiteLoader/include/llapi/EventAPI.h]: maybe we need some tricks to keep ABI stability here

**Comment by KawaiiNahida** [comment]
btw, it's recommend to split those unrelated fixes or feature into separated pull requests

File: LiteLoader/include/llapi/EventAPI.h (hunk 1 above)
Comment (cryx233) [reply LiteLoader/include/llapi/EventAPI.h]: Can we just undo this changes?  It seems that the damage values ​​are all integers.

**Comment by cryx233** [comment]
> btw, it's recommend to split those unrelated fixes or feature into separated pull requests

Ok I will pay attention about this next time

**Comment by Jaciezyt** [comment]
Hmm, you'd better create 3 PRs

**Comment by cryx233** [comment]
will be divided into 3 pr

**Pull request closed without merging**